content/800-data-platform/06-query-console.mdx

```diff
@@ -0,0 +1,68 @@
+---
+title: 'Query Console'
+metaTitle: 'Query Console'
+metaDescription: 'Query Console'
+toc: true
+tocDepth: 3
+---
+
+<TopBlock>
+
+Use the Query Console to run the 3 pre-defined `findMany()` test queries or add and test queries of your own using the [Prisma Client API for model queries](/reference/api-reference/prisma-client-reference#model-queries).
+
+</TopBlock>
+
+## UI components
+
+The Query Console is comprised of 2 major UI components - **Query Editor** and **Query Results Console**.
+
+<img src="https://res.cloudinary.com/prismaio/image/upload/v1657148399/docs/pdp/pdp-query-console-components_w98dp5.png" alt="Query Console - UI Components"/>
+
+| UI component          | Description                                                                         |
+| --------------------- | ----------------------------------------------------------------------------------- |
+| Query Editor          | • Add, edit, delete queries using the Prisma Client API<br />• Run a selected query |
+| Query Results Console | View the results from the selected query                                            |
+
+## Default model queries
+
+- `findMany()`
+- `findMany()` in which the `post_id` is 0
+- `findMany()` but return the data from only 2 fields
+
+## Run a query
+
+You can test and run queries for your Prisma app from the Query Console and view their results on the right side of the screen.
+
+**Steps**
+
+1. In the Query Editor, place the cursor anywhere in a query block.
+2. Run the query using one of the 3 available method.
+   - Click the **Run active query** button in the upper-left.
+   - Click the **Play** (**Run query**) button at the left margin of the editor.
+   - Press Cmd + Enter (on macOS) or Ctrl + Enter (on Windows).
+   <img src="https://res.cloudinary.com/prismaio/image/upload/v1657148832/docs/pdp/pdp-query-console-run-query_lgc5nb.png" width="450px" alt="Query Console - Run a query"/>
+
+**Result**
+
+The query results appeare in the right side in the Query Results Console.
+
+## Add a query
+
+The Query Console supports the autocompletion you get when you use the Prisma Client API in Visual Studio Code.
+
+You can use the same autompletion to create and add new queries.
+
+**Steps**
+
+1. Add new lines after an existing query.
+2. Define your query. Autocomplete is available as you type the available Prisma models and queries based on the generated Prisma Client for the app.
+   <img src="https://res.cloudinary.com/prismaio/image/upload/v1657149670/docs/pdp/schema-add-query-autocomplete_nwkkwf.png" width="500px" alt="Query Console - Add new query and use autocomplete"/>
+
+**Result**
+
+* After you define a query, the green button is available for testing the query.
+* The Prisma Data Platform runs on a best-guess basis even if the query code is not completely defined. For example, in the following snippet, the `select` statement is incomplete, but the Prisma Data Platform still runs the `findMany()` query.
```

Comment:
How does it do that?

Reply:
That's what I found while testing.

Reply:
Tested with a couple of very badly formed queries. `findMany` still runs.

Reply:
Not sure if bug or feature honestly. Sounds a bit scary. 
(For example with a `deleteMany` where the `WHERE` is misspelled - and then it deletes _everything_)

Reply:
FYI I have removed the information about the best-effort because it seems to open a can of worms. I can address it later. Tracking in this docs issue:
https://github.com/prisma/docs/issues/3533